MOB-3106: XCFramework Instructions (#144)
signalfx/splunk-otel-ios: frameworkInstructions.md

> # How to Build an XCFramework

Reviewer: ```suggestion
# How to build an XCFramework
```

> If you would like to import Splunk Otel as a framework into your project, follow these steps:

Reviewer: ```suggestion
If you want to import Splunk Otel as a framework into your project, follow these steps:
```

> ## 1. Ensure the correct build settings.

Reviewer: ```suggestion
## 1. Check build settings
```

> Clone the splunk-otel-ios repo onto your local environment and open the SplunkRumWorkspace.xcworkspace file in XCode. Navigate to the Build Settings tab on the SplunkOtel target and ensure the following two settings are set:
> * Skip Install: No
> * Build Libraries for Distribution: Yes

Reviewer: ```suggestion
Clone the `splunk-otel-ios` repo and open the `SplunkRumWorkspace.xcworkspace` file in Xcode. Navigate to the **Build Settings** tab on the `SplunkOtel` target and make sure the following settings are present:

* Skip Install: No
* Build Libraries for Distribution: Yes
```

> ## 2. Make Archives

Reviewer: ```suggestion
## 2. Make archives
```

> Open the terminal and navigate to the SplunkRumWorkspace/SplunkRum directory. You should be in the same directory as the SplunkRum.xcodeproj file. Paste the following code into the terminal.
> 
>     xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS" -archivePath "archives/SplunkRum-iOS"
> 
> This should create a new archives directory with the file `SplunkRum-iOS.xcarchive` inside it.
> 
> Now repeat the process for the simulator platform:
> 
>     xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS Simulator" -archivePath "archives/SplunkRum-iOS_Simulator"

Reviewer: ````suggestion
Open the terminal and navigate to the directory where the `SplunkRum.xcodeproj` file is located, for example `SplunkRumWorkspace/SplunkRum`. 

Run the following command to create a new archives directory with the file `SplunkRum-iOS.xcarchive` inside:

```bash
    xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS" -archivePath "archives/SplunkRum-iOS"
```

Repeat the process for the simulator platform:

```bash
    xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS Simulator" -archivePath "archives/SplunkRum-iOS_Simulator"
```
````

    
## 3. Make XCFramework

> Paste the following code into the terminal:

Reviewer: ```suggestion
Run the following command to create a new archives directory with the file `SplunkRum-iOS.xcarchive`:
```

>     xcodebuild -create-xcframework -archive archives/SplunkRum-iOS.xcarchive -framework SplunkOtel.framework -archive archives/SplunkRum-iOS_Simulator.xcarchive -framework SplunkOtel.framework -output xcframeworks/SplunkOtel.xcframework
> 
> This will create a new xcframeworks directory with the SplunkOtel.xcframework file inside of it.

Reviewer: ````suggestion
```bash
    xcodebuild -create-xcframework -archive archives/SplunkRum-iOS.xcarchive -framework SplunkOtel.framework -archive archives/SplunkRum-iOS_Simulator.xcarchive -framework SplunkOtel.framework -output xcframeworks/SplunkOtel.xcframework
```
````

> ## 4. Importing the XCFramework into your project
> 
> Open your project in XCode and drag and drop the SplunkOtel.xcframework into the project navigator. This should automatically import the framework. 

Reviewer: ```suggestion
## 4. Import the XCFramework into your project

Open your project in Xcode and drag the `SplunkOtel.xcframework` file into the project navigator. This automatically imports the framework. 
```
